src/mon/MDSMonitor.cc

```diff
@@ -2312,6 +2309,9 @@ void MDSMonitor::tick()
 
   auto &pending = get_pending_fsmap_writeable();
 
+  /* batch any changes to pending with any changes to osdmap */
+  paxos.plug();
```

Comment:
I'm new to this pugging thing - Paxos.h says:

```
  /**
   * true if we want trigger_propose to *not* propose (yet)
   */
  bool plugged = false;
```

That comment is itself confusing for me :)

Could you explain a bit about paxos (un)plug or if its explained anywhere @batrick?

Reply:
It's a relatively new thing. Its purpose revolves around preventing various code paths from triggering a paxos commit before all desired pending state is ready. For us, that mostly means preventing early commit for (1) pending_metadata which is fixed in another PR, and (2) any changes to the osdmap that should be committed with the fsmap changes.

Reply:
Got it. Thanks for explaining.